### PR TITLE
qscintilla2: fix build for Linux

### PR DESCRIPTION
**Pull request by danielnachun**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/qscintilla2.rb [hunk 1]
```diff
@@ -29,9 +29,14 @@ class Qscintilla2 < Formula
   patch :DATA
 
   def install
-    spec = (ENV.compiler == :clang) ? "macx-clang" : "macx-g++"
-    spec << "-arm64" if Hardware::CPU.arm?
-    args = %W[-config release -spec #{spec}]
+    args = "[]"
```
Comment (SeekingMeaning) [inline Formula/qscintilla2.rb:32]:
```suggestion
    args = []
```

**Comment by carlocab** [comment]
You have a merge conflict; you'll need to rebase. Also, I think we can simplify

https://github.com/Homebrew/homebrew-core/blob/a567617d9592796a0008f3532f38f94798624dd8/Formula/qscintilla2.rb#L40-L41

to

```ruby
s.change_make_var! "QMAKE_POST_LINK", "install_name_tool -id #{lib}/$(TARGET1) $(TARGET)"
```

File: Formula/qscintilla2.rb [hunk 2]
```diff
@@ -26,19 +26,22 @@ class Qscintilla2 < Formula
   depends_on "sip"
 
   def install
-    # TODO: when qt 6.1 is released, modify the spec
```
Comment (carlocab) [inline Formula/qscintilla2.rb:29]: Let's keep this comment; it might be useful.

**Comment by carlocab** [comment]
```
Error: An exception occurred within a child process:
  Utils::Inreplace::Error: inreplace failed
qscintilla.pro:
  expected to change "QMAKE_POST_LINK" to "install_name_tool -id /opt/homebrew/Cellar/qscintilla2/2.12.1/lib/$(TARGET1) $(TARGET)"
```

Whoops. I might be confused about how `change_make_var!` works. I guess I'll just fix this another time. Sorry!

**Comment by carlocab** [comment]
Ah, this is the problem. `change_make_var!` expects the variable to be defined at the start of a line:
```ruby
def change_make_var!(flag, new_value)
  return if gsub!(/^#{Regexp.escape(flag)}[ \t]*[\\?+:!]?=[ \t]*((?:.*\\\n)*.*)$/, "#{flag}=#{new_value}", false)

  errors << "expected to change #{flag.inspect} to #{new_value.inspect}"
end
```
This is how `QMAKE_POST_LINK` appears in `src/qscintilla.pro`:
```make
macx:!CONFIG(staticlib) {
    QMAKE_POST_LINK += install_name_tool -id @rpath/$(TARGET1) $(TARGET)
}
```

**Review by carlocab — APPROVED** [review]
The `dynare` failure looks spurious. (Execution timeout; that happens occasionally.)

**Pull request merged**